src/app/content/studyGuides/hooks/openStudyGuides.ts

```diff
@@ -1,21 +1,39 @@
 import { loggedOut } from '../../../auth/selectors';
+import { push } from '../../../navigation/actions';
+import { locationState } from '../../../navigation/selectors';
+import { getQueryForParam } from '../../../navigation/utils';
 import { ActionHookBody } from '../../../types';
-import { actionHook } from '../../../utils';
+import { actionHook, assertNotNull } from '../../../utils';
+import { modalQueryParameterName } from '../../constants';
+import { content } from '../../routes';
 import * as selectContent from '../../selectors';
+import { getBookPageUrlAndParams } from '../../utils';
 import { archiveTreeSectionIsChapter, findArchiveTreeNode } from '../../utils/archiveTreeUtils';
 import { loadMoreStudyGuides, openStudyGuides, setDefaultSummaryFilters } from '../actions';
+import { modalUrlName } from '../constants';
 import * as select from '../selectors';
 
 export const hookBody: ActionHookBody<typeof openStudyGuides> = (services) => async() => {
   const state = services.getState();
   const filtersHaveBeenSet = select.filtersHaveBeenSet(state);
   const currentFilters = select.summaryLocationFilters(state);
   const defaultFilter = select.defaultLocationFilter(state);
+  const currentParams = assertNotNull(selectContent.contentParams(state), 'Opened modal before location was processed');
   const notLoggedIn = loggedOut(state);
-
-  const book = selectContent.book(state);
+  const currentLocationState = locationState(state);
+  const { book, page } = selectContent.bookAndPage(state);
   const firstChapter = book && findArchiveTreeNode(archiveTreeSectionIsChapter, book.tree);
 
+  services.dispatch(push({
```

Comment:
I feel like we should have navigation util that creates push action for specific modal so we don't have to copy pase this part of a code for every modal